tools/vegan_rarefaction/vegan_rarefaction.xml

```diff
@@ -0,0 +1,114 @@
+#end if
+sprintf("Using sample size: %i", raremax)
+Srare <- rarefy(input_abundance, sample=raremax, se=FALSE, MARGIN=1) ##${estimate_standard_error} ##${margin}
+##SHOULD we t(Srare) for out?
+write.table(Srare, "${output_richness}", col.names=NA, sep = "\t")
+
+write.table(rareslope(input_abundance, sample=raremax+raremax_offset), "${output_slope}", sep = "\t")
+
+S <- specnumber(input_abundance)
+
+write.table(S, "${ output_species_count }", col.names=NA, sep="\t" )
+write.table(specnumber(input_abundance, MARGIN=2), "${ output_species_frequency }", col.names=NA, sep="\t" )
+
+write.table(drarefy(input_abundance, sample=raremax+raremax_offset), "${ output_species_probabilities }", col.names=NA, sep="\t" )
+
+##Draw the Curve
+png('${output_plot}', width=1020, height=800, units='px')
+rarecurve(input_abundance, step = ${step_size}, sample = raremax, xlab = "${xlab}", ylab = "${ylab}", label="${label}", col = "blue", cex = 0.6)
+invisible(dev.off())
+
+    ]]>
+        </configfile>
+    </configfiles>
+    <inputs>
+        <param name="input_abundance" type="data" format="tabular" label="File with abundance values for community" help="Rows are samples; columns are species/phyla/community classifier"/>
+        
+        <!-- spec column label?, counts (spec in rows) single select
+        treatment columns multi select, 1+
+        -->
+        
+        <param name="species_column" label="Group name column" type="data_column" data_ref="input_abundance" value="6" help="Species, phylum, etc"/>
+        <param name="sample_columns" label="Sample count columns" type="data_column" multiple="True" value="2" data_ref="input_abundance" help="Select each column that contains counts"/>
+        <!-- <param name="swap_columns_rows" type="boolean" truevalue="true" falsevalue="false" checked="False" label="Swap rows and columns" help="Rows are species/phyla/community classifier; columns are samples"/> -->
+        <param name="sample_size" type="integer" value="" min="1" optional="True" label="Subsample size for rarefying community" help="leave empty to subsample by community size for the smallest sample"/>
+        <!-- <param name="estimate_standard_error" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="False" label="Estimate standard errors"/> -->
+        <!-- <param name="margin" type="integer" value="1" label="Margin for which the index is computed" /> -->
+        
+        <param name="step_size" type="integer" value="1" min="1" label="Step size for sample sizes"/>
+        <param name="xlab" type="text" value="Sample Size" label="X-axis label"/>
+        <param name="ylab" type="text" value="Species" label="Y-axis label"/>
+        <param name="label" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="True" label="Label rarefaction curves by rownames of X"/>
+
+        
+    </inputs>
+    <outputs>
+        <data format="tabular" name="output_species_count" label="${tool.name} on ${on_string} (number of species)"/>
```

Comment:
Are all outputs always needed? If not can we add a select box to choose which outputs I want to have?